### PR TITLE
NuttX: download gcc from https://developper.arm.com

### PR DESCRIPTION
**Pull request by jlecoeur**
Fixes build with -Wimplicit-fallthrough enabled in https://github.com/PX4/Firmware/pull/8551

### CODE REVIEW

File: docker/px4-dev/Dockerfile_nuttx [hunk 1]
```diff
@@ -26,10 +26,10 @@ RUN apt-get update \
 
 # GNU Arm Embedded Toolchain: 7-2017-q4-major December 18, 2017
 RUN mkdir -p /opt/gcc && cd /opt/gcc \
-	&& wget -qO- "https://armkeil.blob.core.windows.net/developer/Files/downloads/gnu-rm/7-2017q4/gcc-arm-none-eabi-7-2017-q4-major-linux.tar.bz2" | tar jx --strip 1 \
+	&& wget -qO- https://developer.arm.com/-/media/Files/downloads/gnu-rm/7-2017q4/gcc-arm-none-eabi-7-2017-q4-major-linux.tar.bz2 | tar jx --strip 1 \ 
 	&& rm -rf /opt/gcc/share/doc
 
-ENV PATH="$PATH:/opt/gcc/bin"
+ENV PATH="/opt/gcc/bin:$PATH"
```
Comment (dagar) [inline docker/px4-dev/Dockerfile_nuttx]: PATH is intentionally prepended for ccache.

**Comment by dagar** [comment]
Huh? Rather concerning that this accounts for the implicit fallthrough nonsense. Any idea why?

If you're going to change the PATH can you prepend /usr/lib/ccache again?

File: docker/px4-dev/Dockerfile_nuttx (hunk 1 above)
Comment (jlecoeur) [reply docker/px4-dev/Dockerfile_nuttx]: As the special fallthrough comments have to be parsed, I thought the problem was due to a mismatch between the pre-compiler and compiler that cmake finds, that is why I put gcc path first. Let me try without this.

Comment (dagar) [reply docker/px4-dev/Dockerfile_nuttx]: I think I'm missing something then. What are you seeing that happens with a compiler other than the one cmake finds?

Comment (jlecoeur) [reply docker/px4-dev/Dockerfile_nuttx]: I can confirm that reverting this change brings the warning back.

Comment (jlecoeur) [reply docker/px4-dev/Dockerfile_nuttx]: Here is the build log: https://pastebin.com/raw/Q8syW4RV

The only visible difference is the location of detected assembler: `/opt/gcc/bin/arm-none-eabi-gcc` vs `/usr/lib/ccache/arm-none-eabi-gcc`

Comment (jlecoeur) [reply docker/px4-dev/Dockerfile_nuttx]: The warning is also back when ccache is prepended like this `ENV PATH="/usr/lib/ccache:/opt/gcc/bin:$PATH"`

**Comment by jlecoeur** [comment]
@dagar It seems related to ccache, /usr/lib/ccache must come after /opt/gcc/bin in PATH, but I have to say I do not understand why.

File: docker/px4-dev/Dockerfile_nuttx (hunk 1 above)
Comment (dagar) [reply docker/px4-dev/Dockerfile_nuttx]: Can you try with `export CCACHE_RECACHE=1`.

**Comment by jlecoeur** [comment]
@dagar I will close here because the issue with -Wimplicit-fallthrough in https://github.com/PX4/Firmware/pull/8551#pullrequestreview-88477261 is not due to the docker image finally.
(BTW this change to the makefile https://github.com/PX4/containers/pull/104/commits/0945c377e4e514114ba0833343f08d7c18717476 was useful to sort locally built images from downloaded images).

Steps to replicate: this builds locally:
```
cd Firmware/
git checkout pr-nuttx_gcc7_test
rm build/ -r
PX4_DOCKER_REPO=px4io/px4-dev-nuttx:2017-12-30 ./Tools/docker_run.sh 'make px4fmu-v2_default'
rm build -r
PX4_DOCKER_REPO=px4io/px4-dev-nuttx:pr-nuttx_gcc7 ./Tools/docker_run.sh 'make px4fmu-v2_default'
```
but does not build on jenkins:
http://ci.px4.io:8080/blue/organizations/jenkins/Firmware/detail/pr-nuttx_gcc7_test/1/pipeline/37/

**Pull request closed without merging**